ckan/lib/search/__init__.py

```diff
@@ -199,23 +155,16 @@ def rebuild(package_id: Optional[str] = None,
     package_index = index_for(model.Package)
     context: Context = {
         'ignore_auth': True,
-        'validate': False,
-        'use_cache': False
     }
 
     if package_id:
-        pkg_dict = logic.get_action('package_show')(context, {
-            'id': package_id
-        })
-        log.info('Indexing just package %r...', pkg_dict['name'])
-        package_index.remove_dict(pkg_dict)
-        package_index.insert_dict(pkg_dict)
+        log.info('Indexing just package %r...', package_id)
+        logic.index_remove_package(package_id)
+        logic.index_update_package(context, package_id)
```

Comment:
We are calling `remove` and then `update`. Does this mean that `update` also creates? Or should we call to `create`?

Reply:
Just me blindly following the old code. I've dropped the `remove` to match the behaviour below.